### PR TITLE
Add SafeEncoder class to handle non-JSON-safe objects

### PR DESCRIPTION
**Pull request by splch**
### Summary

use a safeencoder to parse any object passed in the job

### Details and comments

uses the __string__ method if present otherwise calls repr

### CODE REVIEW

File: qiskit_ionq/helpers.py [hunk 1]
```diff
@@ -448,6 +448,20 @@ def get_user_agent():
...
+class SafeEncoder(json.JSONEncoder):
+    """
+    Custom JSON encoder that handles non-JSON-safe objects by converting them to strings.
+    """
+
+    def default(self, o):
+        try:
+            return super().default(o)
+        except TypeError:
+            return (
```
Comment (Cynocracy) [inline qiskit_ionq/helpers.py:460]: Can you log the fieldname when this happens?

**Review by Cynocracy — CHANGES_REQUESTED** [review]

File: qiskit_ionq/helpers.py [hunk 2]
```diff
@@ -448,6 +448,20 @@ def get_user_agent():
...
+                str(o) if "__str__" in vars(type(o)) else repr(o)
```
Comment (Cynocracy) [inline qiskit_ionq/helpers.py:461]: would prefer another try catch here, maybe you can set it up as a for loop with a few options for the source of the string?

something like

funcs = [ lambda o: super().default(o), ...]

for func in funcs:
   try:
      return func()
   except:
       log
return 'unknown'

Comment (splch) [reply qiskit_ionq/helpers.py]: great yeah i updated to do that - ill look into adding the field name now

**Comment by Cynocracy** [comment]
LGTM with the changes

**Pull request merged**